### PR TITLE
Autogenerate VHT Area Name, CHP Area Name, and VHT's Village 

### PR DESCRIPTION
**Pull request by kennsippell**
#47 

A few notes about trying to get rid of [this line in gross.ts](https://github.com/medic/cht-user-management/pull/108/files#diff-93a51a116fae565ca730cdce509d40f1dc8cc5b7a111219cbb1892826103929bR13):

I attempted to implement this with config interface like `{ "type": "name", "template": "{{ contact.name }} Area" }` so we can do something like prompt for a CHU name AND consistently append something to the end. I added `{{ this }}` to the templates. This worked fine and I really liked how it separated "generation code" from "validation code" - our validation library is poorly name as it is far from just validation.

This created some place resolution issues which I didn't debug, but also resulted in a new problem where the username was `CHU_community_health_unit`. I didn't like that and also didn't know how to resolve it, so I backed off and went with the approach you see in this PR.

### CODE REVIEW

File: src/config/chis-ke/config.json [hunk 1]
```diff
@@ -266,7 +267,7 @@
           "friendly_name": "CHA Name",
           "property_name": "name",
           "type": "name",
-          "parameter": ["\\(PHO\\)", "\\(CHA\\)"],
+          "parameter": ["\\(\\s*PHO\\s*\\)", "\\(\\s*CHA\\s*\\)"],
```
Comment (kennsippell) [inline src/config/chis-ke/config.json:270]: note this change was just during testing. i saw inputs like Richard Nixon (PHO ) which were not being properly formatted

File: src/routes/files.ts [hunk 2]
```diff
@@ -35,9 +25,9 @@ export default async function files(fastify: FastifyInstance) {
         place.name,
         place.contact.properties.name,
         place.contact.properties.phone,
+        place.userRoles.join(' '),
```
Comment (kennsippell) [inline src/routes/files.ts:28]: just reordering this. i personally prefer having username/password at the very end for easy access

File: README.md [hunk 3]
```diff
@@ -61,14 +61,50 @@ required | boolean | True if the object should not exist without this informatio
...
+| phone     | A valid phone number for the specified locality        | Auto formatting provided by [libphonenumber](https://github.com/google/libphonenumber)          | Two letter country code specifying the locality of phone number (eg. `"parameter": "KE"`)             |
+| none      | None                                                   | None                                                                | None                                                                                                   |
+| gender    | A binary gender (eg. `Male`, `Woman`, `M`)             | Formats to either `Male` or `Female`                                | None                                                                                                   |
+| generated | None. No user inputs.                                  | Uses [LiquidJS](https://liquidjs.com) templates to generate data    | None                                                                                                   | [Details](#TheGeneratedConfigPropertyType)
```
Comment (paulpascal) [inline README.md:72]: not really important but for the anchor link to work it needs to be something like `[Details](#The-Generated-ConfigPropertyType)`

**Review by paulpascal — APPROVED** [review]
All right @kennsippell

**Pull request merged**